### PR TITLE
Modify to use marshmallow 3 and flask 2

### PR DESCRIPTION
**Pull request by cullzie**
Hey @viniciuschiele,

Thanks for creating and maintaining this library :t-rex: 

This PR is to update to the latest version of marshmallow available. 
I looked into this as I was having strange issue with the post_dump method. Basically it was silently swallowing an error in marshmallow 2.* versions. I've tested locally myself and everything I've checked seems to work as expected just the marshmallow syntax has changed in some places so my code needs updating.
With 3.* this issue is fixed as well as a lot more improvements it pulls in.

This change is not backward compatible so would require either a major version release or maybe even a new package in pypi.

I've also extracted the logic from flask `_compat` so we can upgrade to flask v2.0.3

Let me know what you think

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
     author_email='vinicius.chiele@gmail.com',
     description='Flask-IO is a library for parsing Flask request arguments into parameters and for serialization of complex objects into Flask response.',
     keywords=['flask', 'rest', 'parse', 'encode', 'decode', 'request', 'json', 'marshmallow'],
-    install_requires=['flask>=0.10.1', 'python-dateutil>=2.4.2', 'marshmallow>=2.9.1,<3'],
+    install_requires=['flask>=0.10.1,<=1.1.4', 'python-dateutil>=2.4.2', 'marshmallow>=3.0.0,<=3.14.1'],
```
Comment (cullzie) [inline setup.py:13]: Pinning to any version of 3 might be better here
`marshmallow>=3`

Comment (viniciuschiele) [reply setup.py]: I think we can use `marshmallow>=3,<4`, the same for flask `flask>=0.10.1,<3`. What do you think?

Comment (cullzie) [reply setup.py]: Ye we can do that. I noticed that flask v2+ does not support any python version <=3.5 now though and this project is listed as having a max python version of 3.4
Python 3.6 is EOL now though so libraries not supporting any version less than that is something to be expected. 
If people want to use this library on those versions they can pin to v1.14.3

Comment (viniciuschiele) [reply setup.py]: Ok, so let's also not support python version <= 3.5 and force the dependencies to `flask>=2,<3` and `marshmallow>=3,<4`. Do you mind making those changes in this PR?
Once it is merged I will release a `v2`

Comment (cullzie) [reply setup.py]: Ye sure, I'll do that now. Do you have a good way to test all these changes? I've done some regression and the unittests are passing but there are a lot of scenarios I haven't covered yet.

Comment (cullzie) [reply setup.py]: Ye I've done some more testing and its just some marshmallow syntax that needs to be updated on my end. 
The unit tests seem to cover the main functionality of the library already so I'm not sure where I could improve the coverage. Functionality wise there should be no change to flask-io

**Comment by cullzie** [comment]
Hey @viniciuschiele, Should I close this off.

**Comment by viniciuschiele** [comment]
Sorry for the delay in this matter, I'm just pretty busy at work.
I will find some time to merge this.
Thanks

**Pull request merged**